backend/src/test/java/controller/UserControllerTest.java

```diff
@@ -0,0 +1,276 @@
+package controller;
+import com.beaplab.BeaplabEngine.constants.BeapEngineConstants;
+import com.beaplab.BeaplabEngine.controller.AccessGroupController;
+import com.beaplab.BeaplabEngine.controller.UserController;
+import com.beaplab.BeaplabEngine.metadata.AccessGroupDto;
+import com.beaplab.BeaplabEngine.metadata.UserDto;
+import com.beaplab.BeaplabEngine.model.AccessGroup;
+import com.beaplab.BeaplabEngine.model.User;
+import com.beaplab.BeaplabEngine.repository.AccessGroupDao;
+import com.beaplab.BeaplabEngine.repository.UserDao;
+import com.beaplab.BeaplabEngine.repository.base.BaseRepository;
+import com.beaplab.BeaplabEngine.service.AccessGroupService;
+import com.beaplab.BeaplabEngine.service.IncorrectLoginsService;
+import com.beaplab.BeaplabEngine.service.UserService;
+import com.beaplab.BeaplabEngine.util.error.UserAlreadyExistException;
+import com.beaplab.BeaplabEngine.util.objectMapper.AccessGroupMapper;
+import com.beaplab.BeaplabEngine.util.objectMapper.UserMapper;
+import org.json.simple.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static utils.MockFactory.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserControllerTest {
+
+    @InjectMocks
+    private UserController userController;
+
+    @Mock
+    private UserService userService;
+
```

Comment:
it is also possible to mock httpsession with
` @Mock
private HttpSession httpSession;`

Reply:
check out WatchControllerTest.java https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-3/blob/development/backend/src/test/java/controller/WatchControllerTest.java

Reply:
Looks good ! good work!